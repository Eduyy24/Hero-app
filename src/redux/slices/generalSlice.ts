import { createSlice } from '@reduxjs/toolkit'

export interface GeneralState {
  pagesData: PageData[];
  isFullData: boolean;
  dataSell: {};
}

const initialState = {
  pagesData: [],
  isFullData: false,
  dataSell: {}
} as GeneralState

export const generalSlice = createSlice({
  name: 'general',
  initialState,
  reducers: {
    setPagesData: (state, action) => {
      state.pagesData = action.payload
      state.isFullData = true;
    },
    setDataSell: (state, action) => {
      state.dataSell = {...state.dataSell, ...action.payload}
    }
  },
})

export const { setPagesData, setDataSell } = generalSlice.actions

export default generalSlice.reducer