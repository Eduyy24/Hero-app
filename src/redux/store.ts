import { configureStore } from '@reduxjs/toolkit'
import thunk from 'redux-thunk';
import storage from 'redux-persist/lib/storage'

import { persistReducer, persistStore } from 'redux-persist';
import generalReducer from './slices/generalSlice'


const persistConfig = {
  key: 'root',
  storage,
}

const persistedReducer = persistReducer(persistConfig, generalReducer)

const store = configureStore({
  reducer: {
    general: persistedReducer
  },
  devTools: process.env.NODE_ENV !== 'production',
  middleware:[thunk]
})

const persistor = persistStore(store);

export { store, persistor };
