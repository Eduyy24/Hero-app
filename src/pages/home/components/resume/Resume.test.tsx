import { render, screen } from "@testing-library/react";
import { Provider } from "react-redux";
import { setDataSell, setPagesData } from "../../../../redux/slices/generalSlice";
import { store } from "../../../../redux/store";
import {Resume} from './Resume';


describe("<ContentHome />", () => {
  test("Render component, and render number row correctly", () => {
    store.dispatch(setPagesData([
      {
        order: 1,
        key: 'step-1',
        fields: [
          {
            type: 'string',
            component: 'inputText',
            key: 'name',
            label: 'Nombre y apellidos',
            rules: {
              required: { value: true, message: 'El campo es obligatorio' },
            }
          }
        ]
      },
    ]))
    store.dispatch(setDataSell({
      name: 'Eduardo Pinedo'
    }))

    render(
      <Provider store={store}>
        <Resume />
      </Provider>
    );

    const content = screen.getByTestId('container-resume')
    expect(content).toMatchSnapshot();
    expect(screen.getByText('Nombre y apellidos:')).toBeInTheDocument()
    expect(screen.getByText('Eduardo Pinedo')).toBeInTheDocument()
    expect(screen.getAllByTestId('list-resume').length).toEqual(1)
  });
});