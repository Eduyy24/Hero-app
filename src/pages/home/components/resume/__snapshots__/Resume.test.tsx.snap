// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<ContentHome /> Render component, and render number row correctly 1`] = `
<div
  data-testid="container-resume"
>
  <h3
    class="title"
  >
    Resumen de venta
  </h3>
  <div
    class="row"
    data-testid="list-resume"
  >
    <strong>
      Nombre y apellidos
      :
    </strong>
    <span>
      Eduardo Pinedo
    </span>
  </div>
  <div
    class="containerButton"
  >
    <button
      class="generalButton"
      name="Finalizar"
    >
      Finalizar
    </button>
  </div>
</div>
`;
